### PR TITLE
Feature/vmo 4326/reenable ivr

### PR DESCRIPTION
**Pull request by safydy-r**
[Overview]
vmo-4326 - flow-builder - PR

[For QA]
The file description in flow-builder would be just a placeholder to simulate the scenario, so don't worry if you see `{a description from backend}`

### CODE REVIEW

File: vue.config.js [hunk 1]
```diff
@@ -41,41 +40,63 @@ module.exports = {
...
         const now = new Date()
           .toISOString()
           .split('T')
 
+        const audio_uuid = `${Math.random()
+          .toString(36)
+          .substr(2, 16)}.${Math.random()
+          .toString(36)
+          .substr(2, 10)}`
+        // TODO (Nice to have): replace original_extension with `req.body.flowFilename.split('.').pop()` when we know how to parse form-data from POST request
```
Comment (safydy-r) [inline vue.config.js:77]: I spent 3h to use `req.body` when we pass form-data from frontend, but didn't success. For now, I'm putting this TODO here as Nice to have. It is not worth spending much more hours I think.

File: src/lib/flow-2.0.0.js [hunk 2]
```diff
@@ -0,0 +1,1358 @@
+/**
```
Comment (safydy-r) [inline src/lib/flow-2.0.0.js:1]: Using the same version of flowjs as in voto5 to make sure both frontend & backend are compatible

File: src/components/common/AudioLibrarySelector.vue [hunk 3]
```diff
@@ -42,25 +42,33 @@ export const AudioLibrarySelector = {
   ],
 
   computed: {
-    selectable() {
-      return !isEmpty(this.alternateSelections)
+    ...mapGetters(['availableAudiosByUri']),
+    selectedFile() {
+      return this.availableAudiosByUri[this.selectedAudioUri]
     },
+    // selectable() {
```
Comment (jcbashdown) [inline src/components/common/AudioLibrarySelector.vue:49]: Remove this comment I think

File: src/components/common/AudioLibrarySelector.vue [hunk 4]
```diff
@@ -26,14 +26,14 @@
 
 <script lang="js">
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types,@typescript-eslint/strict-boolean-expressions */
-import {isEmpty} from 'lodash'
-import {mapActions} from 'vuex'
+// import {isEmpty} from 'lodash'
```
Comment (jcbashdown) [inline src/components/common/AudioLibrarySelector.vue:29]: Remove this

File: src/components/common/AudioLibrarySelection.vue [hunk 5]
```diff
@@ -62,11 +69,19 @@ import momentFilters from '@/lib/filters/moment'
 export const AudioLibrarySelection = {
 
   mixins: [lang, momentFilters],
-  props: ['audioFile', 'selected', 'selectable', 'langId'],
+  props: [
+    'audioFile',
+    // 'selected',
```
Comment (jcbashdown) [inline src/components/common/AudioLibrarySelection.vue:74]: Remove

File: src/components/common/AudioLibrarySelection.vue [hunk 6]
```diff
@@ -26,19 +26,26 @@
...
-            target="_blank"
-            class="tree-block-audio-files-download-original">{{ 'flow-builder.original-file'|trans }}</a>
+            target="_blank">
+            {{ 'flow-builder.original-file' | trans }}
+          </a>
         </li>
 
-        <!-- <li><a :href="`/audiofiles/download/${audioFile.filename}/ogg`" target="_blank" class="tree-block-audio-files-download-ogg">{{'flow-builder.download-X-format'|trans({kind: '.ogg'})}}</a></li>
-        <li><a :href="`/audiofiles/download/${audioFile.filename}/ul`" target="_blank" class="tree-block-audio-files-download-ul">{{'flow-builder.download-X-format'|trans({kind: '.ul'})}}</a></li> -->
+        <!--
```
Comment (jcbashdown) [inline src/components/common/AudioLibrarySelection.vue:34]: Any reason to keep all this?

File: docs/BUILDER_INTEGRATION.md [hunk 7]
```diff
@@ -202,6 +202,12 @@ When using the builder in an app with server side persistence, the following rou
 
 Auth is up to the implementer for now
 
+### Implement a server for the upload route `/audiofiles/upload`
```
Comment (jcbashdown) [inline docs/BUILDER_INTEGRATION.md:205]: Just wondering why the existing dev server wasn't sufficient? Could you also link to the documentation of the server endpoint contract in `routes.md` for this?

File: docs/routes/openapi/routes.yaml [hunk 8]
```diff
@@ -134,6 +134,145 @@ paths:
             The builder should prevent these client side before we get to that point but that cannot always be done due to race conditions around concurrent use.
           content: {}
       x-codegen-request-body-name: language
+/audiofiles/upload:
```
Comment (jcbashdown) [inline docs/routes/openapi/routes.yaml:137]: Where's the .md generated from this?

File: src/store/trees/trees.js [hunk 9]
```diff
@@ -184,7 +184,6 @@ export default {
     treeUpdatedAt: ({tree}) => tree.updatedAt.date,
 
     supportedModes: ({ui}) => ui.supportedModes || [],
-    defaultModes: ({ui}) => ui.defaultModes || [],
```
Comment (jcbashdown) [inline src/store/trees/trees.js:187]: What's the reason we're removing defaultModes support?

**Comment by jcbashdown** [comment]
Looks good but I do have a few questions before approving. It also looks like there is quite a confusing validation message on the upload field before you upload a file. Looks like the text is relevant to text content only?

File: src/store/trees/trees.js (hunk 9 above)
Comment (bulatgab) [reply src/store/trees/trees.js:187]: This change was made by me.
There was 'remove all default modes' in the description.
I suppose it meant remove them in our voto customization in the `flowBuilder.js` only?
LIke:
```
platformBuilderConfig.ui.defaultModes = []
```
@safydy-r Let's revert that commit (having the defaultModes changes) then

**Review by bulatgab — APPROVED** [review]
Well done 👍

File: src/components/common/AudioLibrarySelector.vue (hunk 4 above)
Comment (safydy-r) [reply src/components/common/AudioLibrarySelector.vue]: removed

**Pull request merged**